libyul/AsmParser.h

```diff
@@ -133,20 +133,26 @@ class Parser: public langutil::ParserBase
 		return r;
 	}
 
-	Block parseBlock();
-	Statement parseStatement();
-	Case parseCase();
-	ForLoop parseForLoop();
+	Block parseBlock(ASTLabelRegistryBuilder& _labelRegistryBuilder);
```

Comment:
Would a member pointer to the `ASTLabelRegistryBuilder` be better than having it as a parameter to all those functions? like it was implemented using `ScopedSaveAndRestore` util in `ReferencesResolver` class.